modules/013-helm/hooks/startup_metrics.go

```diff
@@ -42,29 +42,11 @@ var _ = sdk.RegisterFunc(&go_hook.HookConfig{
 func handleDeprecatedAPIStartup(input *go_hook.HookInput, dc dependency.Container) error {
 	input.MetricsCollector.Expire("helm_deprecated_apiversions")
 
-	cl := dc.GetHTTPClient(d8http.WithInsecureSkipVerify())
-
 	// curl -s --connect-timeout 10 --max-time 10 -k -XGET -H "Authorization: Bearer $(cat /var/run/secrets/kubernetes.io/serviceaccount/token)" "https://prometheus.d8-monitoring:9090/api/v1/query?query=resource_versions_compatibility"
-	promURL := "https://prometheus.d8-monitoring:9090/api/v1/query?query=resource_versions_compatibility"
-	req, err := http.NewRequest("GET", promURL, nil)
-	if err != nil {
-		return err
-	}
-	err = d8http.SetKubeAuthToken(req)
-	if err != nil {
-		return err
-	}
-
-	res, err := cl.Do(req)
-	if err != nil {
-		return err
-	}
-	defer res.Body.Close()
-
-	var response promMetrics
-	err = json.NewDecoder(res.Body).Decode(&response)
+	response, err := makePromRequest(dc)
 	if err != nil {
-		return err
+		input.LogEntry.Warnf("Prometheus request for previous metrics failed: %s", err)
```

Comment:
An error is returned if caused by a failure to speak HTTP (such as a network connectivity problem). It would be best to differentiate somehow whether Prometheus is not working or it is not in the cluster.

Reply:
Don't think we need this case. With any problems with prometheus we have to skip this hook (cause it's not life-important) and continue the execution of Deckhouse